lib/commands/general.js

```diff
@@ -16,10 +16,57 @@ commands.active = async function () {
 /**
  * Close app (simulate device home button). If a duration is given, app will
  * re-open after that many seconds
+ *
+ * https://github.com/appium/appium/issues/7741
+ *
+ * Possible values for `timeout`:
+ * - any positive number of seconds: come back after X seconds, show deprecation warning
+ * - any negative number of seconds: never come back, show deprecation warning
+ * - undefined: come back after the default timeout (defined by WDA), show deprecation warning. After deprecation: never come back
+ * - {timeout: 5000}: come back after 5 seconds
+ * - {timeout: null}, {timeout: -2}: never come back
  */
-commands.background = async function (seconds) {
+commands.background = async function (duration) {
+  let seconds;
+  if (_.isUndefined(duration)) {
+    // TODO: Replace the block after deprecated stuff is removed
+    // seconds = Number.MAX_SAFE_INTEGER;
+    log.warn('commands.background: Application under test will never be restored in the future if no duration is provided. ' +
+             'See https://github.com/appium/appium/issues/7741');
+    seconds = null;
```

Comment:
I feel uncomfortable with all the over-determination of `seconds` here. I'd rather use this logic to build up the endpoint and params, and then act on that below.